### ISSUE TITLE
Multi-segment transformation and deformation motion blur 

### ISSUE THREAD

**Issue by dedoardo** [tangent-opensource/hdBlackbird#85]
The motion blur code in `HdCyclesMesh` uses a fixed number of motion steps `HD_CYCLES_MOTION_STEPS` for both transformation and deformation blur. Deformation blur is also always turned on which could results in additional copies of the mesh as the points are not tested to be time varying before they are sampled and copied to the Cycles mesh. The [code](https://github.com/tangent-opensource/hdcycles/blob/main/plugin/hdCycles/mesh.cpp#L109) mentions that it depends on the Schema PR, maybe we can wait for that to clear up the logic.

Changing it requires a bit of refactoring. Are there possible complications to changing it?

**Comment by bareya** [comment]
Hey @dedoardo I totally agree with you that we should have it optimized. I think the number of steps for the specific object should come from the schema. We definitely should have a look into this.

**Comment by bareya** [comment]
This is very closely related to #122. After we merge usd cycles support we can proceed with better motion blur support.

**Comment by bareya** [comment]
I took the liberty and I renamed this ticket to make it more broad. To tackle it at once it's a bit of effort. I think we can break it down to few extra tickets:

* Transformation motion blur 
* Deformation motion blur

At this moment I am kind of in advanced stage of the implementation of transformation motion blur. For deformation motion blur, we would have to refactor the code in few places. For instance for meshes we would have to take into an account possibility of heaving meshes subdivided. During the implementation of deformation motion blur we could refactor Cycles a bit and implement `HdCyclesStandardAttributeSource`. That class would help us to add standard attributes to resource registry.
